### PR TITLE
Unify input name for LKJCorr and LKJCholeskyCov

### PR DESCRIPTION
**Pull request by junpenglao**
* Change LKJCholeskyCov parameter eta->n, n->p
* Change n->nu for degrees of freedom in Wishart
* Updage doc string in `LKJCholeskyCov`

### CODE REVIEW

**Comment by junpenglao** [comment]
Currently, the parameter name in `LKJCholeskyCov` is inconsistent with `LKJCorr`, which is quite confusing. PR to change the parameter name in `LKJCholeskyCov` (as it is newer).

Also using nu for degrees of freedom in `MvStudentT`, `Wishart`, `WishartBartlett`.

**Comment by aseyboldt** [comment]
Hm. I see your point about consistency. But I don't like the parameter names in LKJCorr. Most other parameters are Greek letters, and using `n` for the shape parameter is really confusing when the dimension is a parameter as well. I took the eta from stan.

**Comment by junpenglao** [comment]
@aseyboldt I like `eta` better as well and was thinking of changing the names in `LKJCorr` to be consistent with `LKJCholeskyCov`. But changing the parameter name in `LKJCorr` will break many old codes.

**Comment by twiecki** [comment]
Agree with @aseyboldt, which implies we should change `LKJCorr`'s parameterization.

**Comment by junpenglao** [comment]
@aseyboldt @twiecki OK I change the parameter names in LKJCorr to match LKJCholeskyCov then. I didnt change the use-case in LKJ.ipynb as I understand @aseyboldt you are currently working on it?

**Comment by twiecki** [comment]
We definitely would need a deprecation warning for LKJCorr.

**Comment by junpenglao** [comment]
@twiecki something like:
```
warn('Parameters in LKJCorr has been rename: shape parameter n -> eta, 
  dimension parameter p -> n. Please double check your initialization.',
  DeprecationWarning, stacklevel = 2)
``` 
?

**Comment by twiecki** [comment]
Looks good. has -> have.

**Comment by twiecki** [comment]
@aseyboldt what do you think about the new var names?

**Comment by aseyboldt** [comment]
I like the new ones much better, but where possible I think we should still keep the other keywords and print a warning if they are used.
Unfortunately we can't do that with `LKJCorr`, as `n` is used in both cases, but with a different meaning. This might beak quite some code. How about getting users to use keyword args for now? We could use the old meaning of the parameter names for positional arguments, but print a warning. The only way to get rid of the warning would then be to use the new names with keyword arguments.  We could get this behaviour by checking the `**kwargs`.

**Comment by aseyboldt** [comment]
@junpenglao Yes, I'm working on the notebook.
nit: tailing spaces and there should be an empty line before section headings, I believe.

**Comment by junpenglao** [comment]
> Unfortunately we can't do that with `LKJCorr`, as `n` is used in both cases, but with a different meaning. This might beak quite some code. How about getting users to use keyword args for now? We could use the old meaning of the parameter names for positional arguments, but print a warning. The only way to get rid of the warning would then be to use the new names with keyword arguments. We could get this behaviour by checking the **kwargs.

That was my initial concern. But looking forward I think the users will use `LKJCholeskyCov` much more often (at least I will for sure), as it is much more numerical stable (good job @aseyboldt). I had quite a few simulations and other parameterizations return singluar matrix error quite often. So in a way it is better to align with `LKJCholeskyCov` which the parameter name makes more sense.

I dont know how to do the **kwargs option - do you have an example?

**Comment by junpenglao** [comment]
Proposal: 
change the `__init__` in both `LKJCorr` and `LKJCholeskyCov` from `def __init__(self, n, eta, ...):` 
into `def __init__(self, eta, n, ...):` 
pro: `LKJCorr('lkj', 1, 5)` runs fine as before, with a DeprecationWarning.
con: `LKJCholeskyCov('lkj', 5, 1)` now write as `LKJCholeskyCov('lkj', 1, 5)`, some recent codes will break.

**Comment by junpenglao** [comment]
OK found a semi-acceptable solution, please review @aseyboldt @twiecki 
Only consequency: `LKJCholeskyCov('lkj', 5, 1)` now write as `LKJCholeskyCov('lkj', 1, 5)`, some recent codes will break. `LKJCholeskyCov('lkj', n=5, eta=1)` is fine.

File: pymc3/distributions/multivariate.py [hunk 1]
```diff
@@ -763,14 +763,20 @@ class LKJCorr(Continuous):
     """
 
     def __init__(self, eta=None, n=None, p=None, transform='interval', *args, **kwargs):
-        if p is not None:
+        if (p is not None) & (n is not None) & (eta is None):
```
Comment (twiecki) [inline pymc3/distributions/multivariate.py]: `&` -> `and`

File: pymc3/distributions/multivariate.py [hunk 2]
```diff
@@ -763,14 +763,20 @@ class LKJCorr(Continuous):
...
+            warnings.warn('Reasigning input argument: shape parameter n -> eta; '
```
Comment (twiecki) [inline pymc3/distributions/multivariate.py]: "Reassigning". Also, I would start with: "Parameters to LKJCorr have, changed: shape parameter ... Please update your code. Automatically re-assigning parameters for backwards compatibility."

**Comment by twiecki** [comment]
@junpenglao Given the bad options, I think this is the best one. @aseyboldt?

File: pymc3/distributions/multivariate.py (hunk 2 above)
Comment (junpenglao) [reply pymc3/distributions/multivariate.py]: OK thanks!

**Comment by aseyboldt** [comment]
I think so, too.

**Comment by twiecki** [comment]
Thanks @junpenglao!

**Comment by junpenglao** [comment]
@twiecki @aseyboldt OK I installed from master and the test runs fine - however it's not raising the DeprecationWarning warnings... Maybe we should change it to UserWarning so it's more visible.

**Pull request merged**